### PR TITLE
Update the Philippines holidays: add `en_PH`, `en_US`, `fil`, `th` localization

### LINKED ISSUE #2067: Add Filipino Localization for the Philippines Holiday Class
The `vacanza/holidays` package currently supports public holidays for the Philippines, as defined in the [holidays/countries/philippines.py](https://github.com/vacanza/holidays/blob/1e81ae88f0d12870962cb498006c3f20e4a3d335/holidays/countries/philippines.py) file. However, the package lacks Filipino translations for the holiday names.  It seems we should use `fil` ISO 639-2 code as described [here](https://en.wikipedia.org/wiki/Filipino_language) and [here](https://en.wikipedia.org/wiki/Languages_of_the_Philippines).

The `holidays` l10n documentation section -- https://holidays.readthedocs.io/en/latest/contributing.html#localization

Some l10n PR examples:
  - https://github.com/vacanza/holidays/pull/946
  - https://github.com/vacanza/holidays/pull/1341
  - https://github.com/vacanza/holidays/pull/2060
  - [More PRs](https://github.com/vacanza/holidays/pulls?q=is%3Apr+is%3Aclosed+label%3Al10n) w/ `l10n` label

Personally I would prefer to see `fil_PH` as the default but `en_PH` would also work.

@code-leen we'd appreciate your help with this a lot!

### PR DESCRIPTION
**Pull request by PPsyrius**
## Proposed change

- Adds **`en_PH`**, `en_US`, `fil`, `th` localization for the Philippines holidays.
- Update and Refractor test cases so that the end year is now 2050 like other entries.
- "Ninoy Aquino Day" is now only added from 2004 onwards.

I'm not satisfied with Filipino/Tagalog localization myself, so any double-checks would be appreciated here i.e. [I can't find proper words for `Additional special (non-working) day` since even local news use the English term as-is](https://www.net25.com/news/2024-holidays-inilabas-na-ng-malacanang).

Edit 1: Some additional comments from a Filipino friend:
> The last two entries ("Additional special (non-working) day" and "Elections special (non-working) day") can be translated to "Karagdagang Espesyal na Araw (Walang Trabajo)" and  "Araw ng Halalan (Walang Trabajo)" respectively but that's awkward for me that is as the last two are normally written and said in English in all documentation.

Closes #2067 .

## Type of change

- [ ] New country/market holidays support (thank you!)
- [x] Supported country/market holidays update (calendar discrepancy fix, localization)
- [ ] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/pin/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new `holidays` functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/holidays/tree/dev/docs/source

### CODE REVIEW

**Review by KJhellico — COMMENTED** [review]
New localization 👍

**Comment by PPsyrius** [comment]
Looks like [the new 2025 entries got announced](https://www.officialgazette.gov.ph/nationwide-holidays/2025/) as soon as I get to implement code review suggestions lol.

EDSA appears to be permanently removed from 2024 onwards, so we might want to update the code accordingly as well. 👀

**Review by KJhellico — DISMISSED** [review]
LGTM!

File: README.rst [hunk 1]
```diff
@@ -748,8 +748,8 @@ All other default values are highlighted with bold:
    * - Philippines
      - PH
      -
-     -
-     -
+     - **en_PH**, en_US, th, tl
```
Comment (arkid15r) [inline README.rst:751]: We need to figure out `fil` vs `tl` question. I added a comment [here](https://github.com/vacanza/holidays/issues/2067#issuecomment-2451107369). @KJhellico WDYT?

**Review by arkid15r — COMMENTED** [review]
Great stuff, huge PH improvement. I'm not sure about their languages though:

File: README.rst (hunk 1 above)
Comment (PPsyrius) [reply README.rst:751]: I'm fine either way - alternatively, we can add both `fil` and `tl` for now as well 👀

Comment (KJhellico) [reply README.rst]: As `fil` is officially registered by ISO, and its difference from Tagalog is officially declared, I support `fil`.

**Comment by KJhellico** [comment]
Meantime, in the Philippines: [declaring Sunday, July 27, 2025, as a public holiday (special non-working day)](https://www.officialgazette.gov.ph/2024/10/30/proclamation-no-729-s-2024/). 😉

File: README.rst (hunk 1 above)
Comment (PPsyrius) [reply README.rst]: I'll go with `fil` only and remove `tl` then in that case 👍

Comment (arkid15r) [reply README.rst]: > I'll go with `fil` only and remove `tl` then in that case 👍

Sounds good to me. Should it be `fil_PH`?

Comment (KJhellico) [reply README.rst]: Maybe just `fil`? I don't think there are any other `fil` variation...

Comment (arkid15r) [reply README.rst]: I still haven't figured the locales definition/format out properly, tbh. But:

```
cat /etc/locale.gen | grep -i fil
# This file lists locales that you wish to have built. You can find a list
# this file, you need to rerun locale-gen.
# fil_PH UTF-8
```

Comment (PPsyrius) [reply README.rst]: Hmm - I did run them on Poedit earlier just to make sure and `fil` is a legit code, I don't think we need to add `_PH` suffix here
image

Comment (arkid15r) [reply README.rst]: I don't have an opinion on this. Let's assume `fil` is okay for now and improve it later if needed.

Comment (PPsyrius) [reply README.rst]: We can perhaps keep this for future v1 standardization whether if language codes alone are ok or do we want to enforce locale code as well. :eyes:

**Review by arkid15r — APPROVED** [review]
🇵🇭 LGTM

**Pull request merged**